packages/gatsby/cache-dir/register-service-worker.js

```diff
@@ -23,8 +23,14 @@ if (
               if (navigator.serviceWorker.controller) {
                 // At this point, the old content will have been purged and the fresh content will
                 // have been added to the cache.
-                // We set a flag so Gatsby Link knows to refresh the page on next navigation attempt
-                window.GATSBY_SW_UPDATED = true
+                // We call the onServiceWorkerUpdateReady API so users can show update prompts.
+                apiRunner(`onServiceWorkerUpdateReady`, { serviceWorker: reg })
+
+                // If resources failed for the current page, reload.
+                if (window.___failedResources) {
+                  console.log(`resources failed, SW updated - reloading`)
```

Comment:
There might be scenarios where resources don't fail (keeping `public` from previous builds and deploying both new and old files), but we would still want to reload on SW update?

Reply:
I think the idea is we should only reload if necessary to prevent breaking user flow - the new API allows users to force a reload on update if they want to do so.